packages/blocks/github-pr-overview/src/entity-aggregations.ts

```diff
@@ -1,261 +1,297 @@
-    .map((_, pageNumber) =>
-      getPrs(githubPullRequestTypeId, aggregateEntities, accountId, pageNumber),
-    );
-
-  Promise.all(promises)
-    .then((entitiesResults) => {
-      const entities: GithubPullRequest[] = entitiesResults
-        .flatMap((entityResult) => entityResult?.results)
-        .filter(isDefined);
-
-      const pullRequests = uniqBy(entities, "id");
-
-      const mappedPullRequests = new Map();
-
-      for (const pullRequest of pullRequests) {
-        const pullRequestId = `${pullRequest.repository}/${pullRequest.number}`;
-        mappedPullRequests.set(pullRequestId, pullRequest);
-      }
+  if (response.errors || !response.data) {
+    throw new Error("An error occured while fetching events");
+  }
+  const entityTypes = response.data.results;
 
-      setState(mappedPullRequests);
-      if (mappedPullRequests.size === 0) {
-        setBlockState(BlockState.Error);
-      }
-    })
-    .catch((err) => {
-      throw err;
-    });
-};
+  const pullRequestTypeId = entityTypes.find(
+    (entityType) => entityType.schema.title === "GithubPullRequest",
+  )?.entityTypeId;
+  const reviewTypeId = entityTypes.find(
+    (entityType) => entityType.schema.title === "GithubReview",
+  )?.entityTypeId;
+  const issueEventTypeId = entityTypes.find(
+    (entityType) => entityType.schema.title === "GithubIssueEvent",
+  )?.entityTypeId;
 
-const getReviews = (
-  githubReviewTypeId: string,
-  aggregateEntities: BlockProtocolAggregateEntitiesFunction | undefined,
-  accountId: string | null | undefined,
-  pageNumber: number | undefined,
-  selectedPullRequest: PullRequestIdentifier,
-): Promise<BlockProtocolAggregateEntitiesResult<GithubReview> | void> => {
-  if (!aggregateEntities) {
-    return new Promise<void>(() => {});
+  if (pullRequestTypeId && reviewTypeId && issueEventTypeId) {
+    const githubTypeIds: {
+      [key in GITHUB_ENTITY_TYPES]: string;
+    } = {
+      [GITHUB_ENTITY_TYPES.PullRequest]: pullRequestTypeId,
+      [GITHUB_ENTITY_TYPES.Review]: reviewTypeId,
+      [GITHUB_ENTITY_TYPES.IssueEvent]: issueEventTypeId,
+    };
+    return githubTypeIds;
+  } else {
+    throw new Error("An error occured while fetching Github Entity Types");
   }
+};
 
-  const res = aggregateEntities({
-    accountId,
-    operation: {
-      entityTypeId: githubReviewTypeId,
-      pageNumber,
-      itemsPerPage: ITEMS_PER_PAGE,
-      multiFilter: {
-        operator: "AND",
-        filters: [
+export const getPrs = async (
+  githubPullRequestTypeId: string,
+  aggregateEntities: GraphBlockHandler["aggregateEntities"],
+  numPages: number = 5,
+  selectedPullRequest?: PullRequestIdentifier,
+): Promise<GithubPullRequestEntityType[]> => {
+  const response = await aggregateEntities({
+    data: {
+      operation: {
+        entityTypeId: githubPullRequestTypeId,
+        pageNumber: 1,
+        itemsPerPage: ITEMS_PER_PAGE * numPages,
+        multiFilter: {
+          operator: "AND",
+          filters:
```

Comment:
It was honestly just something I did during development to experiment with the interface. I am not sure if we want to possibly get ALL things in one call because that seems like it might be slower as opposed to doing multiple promises 'in parallel', and also I'm a little worried about the implication of requesting a tonne of data in one request if backends end up limiting the size of a request due to their memory constraints or something.

Reply:
Sorry I didn't make this clearer.  I went with the approach of fetching all things because the existing setup did that. I figure we'd want to make this optimal later on by initially fetching a small amount of data and getting more if the user wants that (could be via a load more button or on scroll). Decided to leave that out since it's out of scope for now. 

I didn't consider the fact that multiple api calls in 'parallel' will be faster than one call or backends could limit the sizing of requests. I do feel we'd eventually rewrite this since the current behaviour of fetching "ALL PRs" before a user can use the block isn't a great experience.

Happy to revert though if you want me to